### ISSUE TITLE
InvalidOperationException related to GetRunStepsAsync

### ISSUE THREAD

**Issue by dspear** [openai/openai-dotnet#203]
### Confirm this is not an issue with the underlying OpenAI API

- [X] This is an issue with the Python library

### Confirm this is not an issue with Azure OpenAI

- [X] This is not an issue with Azure OpenAI

### Describe the bug

I am retrieving the RunSteps from an Assistant. I go through the standard sequence: GetRunStepsAsync, GetAllValuesAsync, await foreach for each step. If there was a file_search step, the foreach generates this exception:
System.InvalidOperationException: 'The requested operation requires an element of type 'String', but the target element has type 'Object''.

This occurs with the latest beta-10 and with the older beta-6 that we have in the field. This is new behavior within the last couple of days, which makes me think that there is a change in the base API that is now triggering this exception.

I don't know if this is an issue with the Python library, but I had to check that to enable the Submit button.

### To Reproduce

1. Make a request via the Assistants API that will use the file_search feature.
2. Get the RunSteps.
This generates System.InvalidOperationException

### Code snippets

```C#
AsyncPageCollection<RunStep> res = _assistantClient.GetRunStepsAsync(threadId, runId, null, ct);
            IAsyncEnumerable<RunStep> values = res.GetAllValuesAsync(ct);

            await foreach(RunStep step in values)
            {
// Whatever, code does not execute because the foreach throws an exception
            }
```

### OS

Windows 11

### .NET version

Framework 4.8

### Library version

2.0.0-beta.10

**Comment by joseharriaga** [comment]
Thank you for reaching out, @dspear ! I'm able to repro the issue. You are correct that the issue is related to a recent change in the service's REST API exposing a latent bug in the library. We're working on a fix that we hope to release early next week.

**Comment by dspear** [comment]
I'm glad you'll be able to fix it in the library. I'm concerned about my product (and other people's perhaps) out in the field. Is there any way to contact the REST API team and see if they can change the API back so it doesn't break the library?

**Comment by dspear** [comment]
One more thing: Can you tell me what the specific change was to the REST API? I have an intermediary proxy that I may be able to use to undo this change, if I know what I'm looking for.

**Comment by joseharriaga** [comment]
This has been fixed as part of this PR: 🔗 https://github.com/openai/openai-dotnet/pull/206

We also just pushed a release today, so you can go grab this fix starting with version 2.0.0-beta.11:
🔗 https://www.nuget.org/packages/OpenAI/2.0.0-beta.11

Thank you!
